Improve mango docs (#639)
apache/couchdb-documentation: src/api/database/find.rst

it easier to take advantage of future improvements to query planning
(e.g. automatic selection of partial indexes).

.. note::

>     An index with fields will never be used, unless the selector includes

Reviewer: Do you think double-inverting this sentence would make it easier to understand?
```suggestion
    An index with fields is only used, when the selector includes
```

>     all of the fields indexed. For instance, if an index contains ``["a". "b"]``

Reviewer: From this example, I understand that
| Selector | Index | Index is used |
|----------|-------|---------------|
| a        | a, b  | no            |
| a, b     | a     | yes           |

Is that correct?
